core/src/main/java/org/infinispan/context/impl/NonTxInvocationContext.java

```diff
@@ -56,6 +57,18 @@ public void putLookedUpEntry(Object key, CacheEntry e) {
                    Collections.emptyMap() : lookedUpEntries);
    }
 
+   @Override
+   public void forEachEntry(BiConsumer<Object, CacheEntry> consumer) {
```

Comment:
most of the places iterate over `!entry.isRemoved() && !entry.isNull()`. 
should we add a method that only iterates over valid entries?

Reply:
Ok, makes sense